src/lib.rs

```diff
@@ -172,6 +172,57 @@ struct Wallet {
     wallet_mutex: Mutex<BdkWallet<AnyDatabase>>,
 }
 
+pub struct OutPoint {
+    txid: String,
+    vout: u32,
+}
+
+pub struct TxOut {
+    value: u64,
+    address: String,
+}
+
+pub enum KeychainKind {
```

Comment:
I'm not sure we need the 0 and 1 here but maybe there is something I'm missing. Can you tell me why you added them and not just use the enum defined in BDK directly? (i.e. no need to define it here in Rust and you could simply have it in the UDL file I think)

Reply:
Follow-up: my [tests pass](https://github.com/thunderbiscuit/bdk-kotlin-test-suite/blob/23527ef3797eff48afc81b11af0fd7c776c84734/app/src/test/kotlin/bdk/kotlin/testsuite/ListUnspent.kt#L35-L46) when refactoring the KeychainKind out of the lib.rs file like so:
```rust
// pub enum KeychainKind {
//     External = 0,
//     Internal = 1,
// }

// impl From<bdk::KeychainKind> for KeychainKind {
//     fn from(x: bdk::KeychainKind) -> KeychainKind {
//         match x {
//             bdk::KeychainKind::External => KeychainKind::External,
//             bdk::KeychainKind::Internal => KeychainKind::Internal,
//         }
//     }
// }

impl NetworkLocalUtxo for LocalUtxo {
    fn from_utxo(x: &bdk::LocalUtxo, network: Network) -> LocalUtxo {
        LocalUtxo {
            outpoint: OutPoint {
                txid: x.outpoint.txid.to_string(),
                vout: x.outpoint.vout,
            },
            txout: TxOut {
                value: x.txout.value,
                address: bdk::bitcoin::util::address::Address::from_script(
                    &x.txout.script_pubkey, network).unwrap().to_string(),
            },
            keychain: x.keychain,
        }
    }
}
```

Reply:
Sorry, my mistake. I initially assumed that was a requirement of uniFFI but then forgot to try without. Thanks for the suggestion!